netx.go

```diff
@@ -101,7 +103,9 @@ func DialContext(ctx context.Context, network string, addr string) (net.Conn, er
 	dialer := dial.Load().(func(context.Context, string, string) (net.Conn, error))
 
 	conn, err := dialer(ctx, network, addr)
```

Comment:
This will always first try the given thing first even if it has discovered a nat64 prefix right?  @benjojo would this typically immediately fail locally since there is no ipv4 routing present locally or would this put packets on the wire?

Reply:
If there are zero IPv4 addresses on the device this should instantly fail. If there is one there that is _broken_ (as might be the case you would have to wait for a

Reply:
Still an improvement as-is, but I wonder if we should only take that penalty once in those cases and try dialing ipv6 first if we've got a cached nat64 prefix.  Downside is, I think it could make it convoluted and might need to have more cases for invalidating the prefix if fails.